src/lib/mina.ts

```diff
@@ -154,11 +160,18 @@ function createTransaction(
   if (feePayerKey !== undefined) {
     // if senderKey is provided, fetch account to get nonce and mark to be signed
     let senderAddress = feePayerKey.toPublicKey();
-    let senderAccount = getAccount(senderAddress, TokenId.default);
+
+    let nonce_;
+    if (!nonce) {
```

Comment:
this should check for `nonce === undefined`, otherwise a `0` nonce is not treated correctly

Reply:
good catch